### PR TITLE
Modularizes and reduces duplication in helm chart

### PR DESCRIPTION
**Pull request by AbhijithGanesh**
Signed-off-by: Abhijith Ganesh <67182544+AbhijithGanesh@users.noreply.github.com>

## Description

Fixes pyrsia/pyrsia#906

Modularizes and reduced duplication in helm chart

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and everything passes.
- [x] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

**Comment by AbhijithGanesh** [comment]
@sbtaylor15 I think you should be added as a reviewer too, I can't add it myself

**Comment by betarelease** [comment]
Based on what is in the PR, I would say the PR title should be

`Modularizes and reduced duplication in helm chart`

` This PR modularizes helm chart for Kubernetes deployment of pyrsia_node`

**Review by betarelease — DISMISSED** [review]
Based on what is in the PR, I would say the PR title should be

Modularizes and reduced duplication in helm chart

This PR modularizes helm chart for Kubernetes deployment of pyrsia_node

**Comment by AbhijithGanesh** [comment]
I'll re name the PR as suggested.

**Comment by AbhijithGanesh** [comment]
@sbtaylor15 I have removed the annotations to metacontroller. Its support is very limited. I have done some research and I can understand that Pyrsia will not require a service at a pod level. A loadbalancer should address issue [906](https://github.com/pyrisa/pyrsia/issues/906) without much hassle when scaled. The load balancer already extends support!

**Comment by sbtaylor15** [comment]
> @sbtaylor15 I have removed the annotations to metacontroller. Its support is very limited. I have done some research and I can understand that Pyrsia will not require a service at a pod level. A loadbalancer should address issue [906](https://github.com/pyrisa/pyrsia/issues/906) without much hassle when scaled. The load balancer already extends support!

We need to have a 1 to 1 Service to Pod in order to give each pod its unique external ip address.  So when a StatefulSet is scaled the corresponding services need to be scaled to maintain the external ip address.  Each libp2p node needs to have its own unique ip address.

File: installers/helm/pyrsia-node/templates/service.yaml [hunk 1]
```diff
@@ -1,64 +1,20 @@
----
 kind: Service
 apiVersion: v1
 metadata:
-  name: pyrsia-node-0
+  name: pyrsia-node
   annotations:
-    kubernetes.io/ingress.global-static-ip-name: pyrsia-node-0
+    kubernetes.io/ingress.global-static-ip-name: pyrsia-node
```
Comment (betarelease) [inline installers/helm/pyrsia-node/templates/service.yaml:6]: Will this service create pyrsia-nodes via the replicaset and will they have unique names like `pyrsia-node-0`?

My k8s knowledge is dated hence I ask. @sbtaylor15

File: installers/helm/pyrsia-node/values.yaml [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+    repository: sbtaylor15/pyrsia-node
+    tag: latest
+    pullPolicy: Always
+
+pyrsia:
+    ports:
+        pyrsia: 80
+        p2p: 44000
+    storage:
+        size: "1500Mi"
```
Comment (betarelease) [inline installers/helm/pyrsia-node/values.yaml:13]: This went from 1500Gi to 1500Mi. Needs to be 1500Gi.

File: installers/helm/pyrsia-node/templates/statefulset.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ spec:
       track: stable
   template:
     metadata:
-      labels:
+        labels:
```
Comment (betarelease) [inline installers/helm/pyrsia-node/templates/statefulset.yaml:18]: fix tab. there are extra 2 spaces here.

File: installers/helm/pyrsia-node/templates/service.yaml (hunk 1 above)
Comment (sbtaylor15) [reply installers/helm/pyrsia-node/templates/service.yaml:6]: @betarelease At this point we need to pre-define a list of static ip address for the service.  Adding [ExternalDNS](https://github.com/kubernetes-sigs/external-dns) will remove the need to pre-define ip addresses minimizing setup and on-going DNS management.  ExternalDNS will be in a separate PR.

File: installers/helm/pyrsia-node/values.yaml (hunk 2 above)
Comment (sbtaylor15) [reply installers/helm/pyrsia-node/values.yaml:13]: Yes, 1500Gi is needed to fit the Docker images and Java artifacts.

Comment (AbhijithGanesh) [reply installers/helm/pyrsia-node/values.yaml:13]: Oops, this was a genuine mistake, I will patch it

File: installers/helm/pyrsia-node/templates/service.yaml (hunk 1 above)
Comment (AbhijithGanesh) [reply installers/helm/pyrsia-node/templates/service.yaml:6]: Can I create an issue about it? Adding external DNS to pre-defined IP Addr

File: installers/helm/pyrsia-node/templates/statefulset.yaml (hunk 3 above)
Comment (AbhijithGanesh) [reply installers/helm/pyrsia-node/templates/statefulset.yaml:18]: Will edit it

File: installers/helm/pyrsia-node/templates/service.yaml (hunk 1 above)
Comment (sbtaylor15) [reply installers/helm/pyrsia-node/templates/service.yaml:6]: See #997

**Review by sbtaylor15 — APPROVED** [review]
/lgtm

**Pull request closed without merging**